radsat-sk/operation/message/crypt/RKey.c

```diff
@@ -0,0 +1,81 @@
+/**
+ * @file RKey.c
+ * @date February 9, 2022
+ * @author Tyrel Kostyk (tck290)
+ */
+
+#include <RKey.h>
+#include <RFram.h>
+
+
+/***************************************************************************************************
+                                   DEFINITIONS & PRIVATE GLOBALS
+***************************************************************************************************/
+
+/** The hardcoded address of the FIRST copy of the private 1-byte key in FRAM memory. */
+#define PRIVATE_KEY_ADDR_ONE	((uint32_t)0x1001AA00)	// TODO: set to real address location
+
+/** The hardcoded address of the SECOND copy of the private 1-byte key in FRAM memory. */
+#define PRIVATE_KEY_ADDR_TWO	((uint32_t)0x1001AA04)	// TODO: set to real address location
+
+/** The hardcoded address of the THIRD copy of the private 1-byte key in FRAM memory. */
+#define PRIVATE_KEY_ADDR_THREE	((uint32_t)0x1001AA08)	// TODO: set to real address location
+
+
+/** The 1-byte private key used for the decryption operations. */
+static uint8_t key = 0;
+
+
+/***************************************************************************************************
+                                             PUBLIC API
+***************************************************************************************************/
+
+/**
+ * Return the Private Key used for decryption, retrieving it from FRAM if necessary.
+ *
+ * Will use a "best 2 out of 3" voting approach to add redundancy in the case of bit flip
+ * errors or memory corruption. If all 3 are different, the first one is return. Retrieval
+ * and voting only done on first access per start-up; stored locally afterwards.
+ *
+ * @return The private key.

```

Comment:
Add to the return status on what happens if there is an error (e.g. if error == 1, then it fails)